client-js/app/elements/labrad-registry.ts

```diff
@@ -206,34 +207,41 @@ export class LabradRegistry extends polymer.Base {
     if (event.ctrlKey || event.button == 2 ) {
       //should I use switch and case here instead of ifs?
       this.$.dragOp.innerText = 'Copy';
+      this.$.dragDialog.open();
+      window.setTimeout(() => this.$.dragPathInput.$.input.focus(), 0);
     } else {
       this.$.dragOp.innerText = 'Move';
+      this.$.dragDialog.open();
+      window.setTimeout(() => this.$.dragPathInput.$.input.focus(), 0);
     }
-    this.$.dragDialog.open();
-    window.setTimeout(() => this.$.dragPathInput.$.input.focus(), 0);
+
   }
 
   /**
    * handles folders dropped not into folders
    */
   @listen('drop')
-  handleDrop(event) {{"path":["two"]}
+  handleDrop(event) {
     event.preventDefault();
     var data = JSON.parse(event.dataTransfer.getData('text'));
     var newPath: string[] = this.path.slice();
     newPath.push(event.target.closest('td').name);
```

Comment:
I get errors on this line:
```
Uncaught TypeError: Cannot read property 'name' of null
```
If the drop happens somewhere else in the registry list (not on a subdirectory) we should just use the current path and not push an additional segment onto it.